Automattic/wp-calypso
# Stats: Fix case where PWYW slider has no initial selection

Related to #84859. Should merge PR #85018 first.

## Proposed Changes

Validates the provided initialValue prop to make sure it's both within the expected range and it's an integer value. Fixes the case where the slider could show with no initial value/emoji visible.

<img width="425" alt="SCR-20231208-qgcj" src="https://github.com/Automattic/wp-calypso/assets/40267301/f123e837-6c05-43dd-b1bc-fa2a7fa73c0f">

## Testing Instructions



* Launch the Calypso Live link.
* Navigate to **Stats → Traffic**.
* Click on CTA to purchase Stats plan.
* Add `stats/tier-upgrade-slider` to URL flags.
* Confirm tier slider shows the amount and the emoji correctly.

I set my currency to JPY to trigger this bug. Depends on the steps computation logic generating an odd number of steps.

## Pre-merge Checklist
